pkg/webhooks/leaderworkerset_webhook_test.go

```diff
@@ -0,0 +1,286 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhooks
+
+import (
+	"github.com/google/go-cmp/cmp"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	utilvalidation "k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"testing"
+)
+
+func TestGetPercentValue(t *testing.T) {
+	tests := []struct {
+		name           string
+		input          intstr.IntOrString
+		wantOutputVal  int
+		wantOutputBool bool
+	}{
+		{
+			name: "input type int",
+			input: intstr.IntOrString{
+				Type:   0,
+				IntVal: 1,
+			},
+			wantOutputVal:  0,
+			wantOutputBool: false,
+		},
+		{
+			name: "input type string - invalid format",
+			input: intstr.IntOrString{
+				Type:   1,
+				StrVal: "1",
+			},
+			wantOutputVal:  0,
+			wantOutputBool: false,
+		},
+		{
+			name: "input type string - valid format",
+			input: intstr.IntOrString{
+				Type:   1,
+				StrVal: "1%",
+			},
+			wantOutputVal:  1,
+			wantOutputBool: true,
+		},
+		{
+			name: "input type string - valid format",
+			input: intstr.IntOrString{
+				Type:   1,
+				StrVal: "101%",
+			},
+			wantOutputVal:  101,
+			wantOutputBool: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			outputVal, outputBool := getPercentValue(tc.input)
+			if diff := cmp.Diff(tc.wantOutputVal, outputVal); diff != "" {
```

Comment:
Nit: No need to use diff here, you can compare simply. Not a big deal, I'm ok to ignore this. Same to other places.

Reply:
keeping diff as I see diff being used at other places as well.